Signal/src/call/OutboundCallInitiator.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+/**
+ * Creates an outbound call via either Redphone or WebRTC depending on participant preferences.
+ */
+@objc class OutboundCallInitiator: NSObject {
+    let TAG = "[OutboundCallInitiator]"
+
+    let callUIAdapter: CallUIAdapter
+    let redphoneManager: PhoneManager
+    let contactsManager: OWSContactsManager
+    let contactsUpdater: ContactsUpdater
+
+    init(redphoneManager: PhoneManager, callUIAdapter: CallUIAdapter, contactsManager: OWSContactsManager, contactsUpdater: ContactsUpdater) {
+        self.redphoneManager = redphoneManager
+        self.callUIAdapter = callUIAdapter
+
+        self.contactsManager = contactsManager
+        self.contactsUpdater = contactsUpdater
+    }
+
+    /**
+     * |handle| is a user formatted phone number, e.g. from a system contacts entry
+     */
+    public func initiateCall(handle: String) -> Bool {
+        Logger.info("\(TAG) in \(#function) with handle: \(handle)")
+        guard let recipientId = PhoneNumber(fromUserSpecifiedText: handle).toE164() else {
+            Logger.warn("\(TAG) unable to parse signalId from phone number: \(handle)")
```

Comment:
If a contact has multiple phone numbers (work cell, home cell, fax, etc.) which one(s) does signal receive?  If more than one, shouldn't we try to parse/lookup all of them and use the first one that corresponds to a signal user?